semeio/jobs/ots/ots.py

```diff
@@ -0,0 +1,597 @@
+        restart file - otherwise an exception will be raised.
+        """
+        restart_view = self._rst_file.restartView(sim_time=date)
+        self.subsidence.add_survey_PRESSURE(name, restart_view)
+
+        self._restart_views[name] = restart_view
+
+        return restart_view
+
+    @staticmethod
+    def _divide_negative_shift(ts):
+        for i, _ in enumerate(ts):
+            if ts[i] < 0:
+                ts[i] /= 5.0
+
+    def geertsma_ts_rporv(self, vintage_pairs):
+        """
+        Calculates TS without using velocity. Fast.
+        Velocity is only used to get the surface on the velocity grid.
+        Uses change in porevolume from Eclipse (RPORV in .UNRST) as input to
+        Geertsma model.
+
+        :param vintage_pairs:
+        """
+
+        if len(vintage_pairs) == 0:
+            return 0, []
+
+        vintages = self._vintages_name_date(vintage_pairs)
+        surface = self._surface
+        num_points = len(surface)
+        num_points_calculated = 0
+        du = np.zeros((len(vintages.name), num_points))
+        ts_surfaces = []
+        for iv, (vn, vd) in enumerate(zip(vintages.name, vintages.date)):
+            if self._verbose:
+                print(
+                    "{:%x %X} TS_RPORV: Calculating vintage {:%Y.%m.%d}".format(
+                        dt.now(), vd
+                    )
+                )
+                sys.stdout.flush()
+
+            self.add_survey(vn, vd)
+
+            num_points_calculated = 0
+            for point in range(num_points):
+
+                if not np.isnan(surface.z[point]):
+                    num_points_calculated += 1
+                    r1 = (surface.x[point], surface.y[point], 0)
+                    r2 = (
+                        surface.x[point],
+                        surface.y[point],
+                        surface.z[point] - self._seabed,
+                    )
+                    # subsidence and displacement have opposite sign
+                    # should have minus on dz1 and dz2 here,
+                    # more efficient when calculating ts
+                    dz1 = self.subsidence.eval_geertsma_rporv(
+                        base_survey=vn,
+                        monitor_survey=None,
+                        pos=r1,
+                        youngs_modulus=self._youngs_modulus,
+                        poisson_ratio=self._poisson_ratio,
+                        seabed=self._seabed,
+                    )
+                    dz2 = self.subsidence.eval_geertsma_rporv(
+                        base_survey=vn,
+                        monitor_survey=None,
+                        pos=r2,
+                        youngs_modulus=self._youngs_modulus,
+                        poisson_ratio=self._poisson_ratio,
+                        seabed=self._seabed,
+                    )
+                    du[iv, point] = dz2 - dz1
+
+        for ivp, vp in enumerate(vintage_pairs):
+            if self._verbose:
```

Comment:
As mentioned, using logger instead would help a bit on this part. And still the amount of logging here kind of scrambles what the function really does. The messages for instance are identical, the thing different is the order of the dates.

Reply:
The logging (as mentioned bellow) can be extracted to helper methods.